### PR TITLE
[build] Fix openssl static linking error     

### PR DESCRIPTION
**Pull request by wangyoucao577**
- Upgrade `OPENSSL_USE_STATIC_LIBS` as an option to not only affects windows, but also `pkg-config` found openssl;    
- Remove useless `--OPENSSL_USE_STATIC_LIBS=TRUE` in android script since it doesn't have any effects;        
- Usage: `PKG_CONFIG_PATH=<YOUR_STATIC_SSL_PKGCONFIG_PATH> cmake .. -DENABLE_STATIC=ON -DENABLE_SHARED=OFF -DUSE_OPENSSL_PC=ON -DOPENSSL_USE_STATIC_LIBS=ON `

More analysis see https://github.com/Haivision/srt/issues/1172#issuecomment-1152849911

Closes https://github.com/Haivision/srt/issues/1172

### CODE REVIEW

**Comment by wangyoucao577** [comment]
Both coverage decrease and the Travis CI build fail don't seem to make sense since changes in this PR doesn't change any code and any default behavior. Anyone can help to check it?

File: CMakeLists.txt [hunk 1]
```diff
@@ -148,6 +148,7 @@ option(ENABLE_CODE_COVERAGE "Enable code coverage reporting" OFF)
 option(ENABLE_MONOTONIC_CLOCK "Enforced clock_gettime with monotonic clock on GC CV" ${ENABLE_MONOTONIC_CLOCK_DEFAULT})
 option(ENABLE_STDCXX_SYNC "Use C++11 chrono and threads for timing instead of pthreads" ${ENABLE_STDCXX_SYNC_DEFAULT})
 option(USE_OPENSSL_PC "Use pkg-config to find OpenSSL libraries" ON)
+option(OPENSSL_USE_STATIC_LIBS "Use static libraries of openssl, affects `USE_OPENSSL_PC` found openssl or on Windows" OFF)
```
Comment (maxsharabayko) [inline CMakeLists.txt:151]:
```suggestion
option(OPENSSL_USE_STATIC_LIBS "Link OpenSSL libraries statically." OFF)
```

**Review by maxsharabayko — CHANGES_REQUESTED** [review]

File: scripts/build-android/mksrt [hunk 2]
```diff
@@ -17,7 +17,7 @@ done
 
 cd $SRC_DIR
 ./configure --use-enclib=$ENC_LIB \
---use-openssl-pc=OFF --OPENSSL_USE_STATIC_LIBS=TRUE \
```
[the remark this answers is not in the document]
Comment (maxsharabayko) [reply scripts/build-android/mksrt:20]: Ok, merging as is. We can have a follow-up fix later if needed. At least Android CI build is passing.

**Comment by maxsharabayko** [comment]
@wangyoucao577 
Thank you for this great PR!
I had to squash-merge it into one commit to avoid having 8 commits addressing the same change. But thank you for following the contribution guide and naming those commits properly!

**Pull request merged**